File: TestAppSpider/app/src/main/java/com/example/sylvius/testappspider/Servo.java
package com.example.sylvius.testappspider;

/**
 * Created by Sylvius on 11-5-2016.
 */
public class Servo {
    private int id;
    private float Position,Load, Voltage, Temperature;
    private int IsMoving;
    private String[] allData = new String[]{"ID: "+id,"Position: "+Position+"","Load: "+Load+"","Temperature: "+Temperature+"","Voltage: "+Voltage+""}; //without IsMoving!

    public Servo(int id, float Position, float Load, float Temperature, float Voltage, int isMoving) {
        this.id = id;
        this.Position = Position;
        this.Load = Load;
        this.Temperature = Temperature;
        this.Voltage = Voltage;
        this.IsMoving = isMoving;
    }

    public Servo(){}

    public int getId() {
        return id;
    }

    public float getPosition() {
        return Position;
    }

    public void setPosition(float position) {
        Position = position;
    }
    public float getLoad(){
        return Load;
    }
    public void setLoad(float load){
        Load = load;
    }
    public float getVoltage() {
        return Voltage;
    }
    public void setVoltage(float voltage) {
        Voltage = voltage;
    }
    public float getTemperature() {
        return Temperature;
    }
    public void setTemperature(float temperature) {
        Temperature = temperature;
    }
    public int isMoving() {
        return IsMoving;
    }

    public void setMoving(int moving) {
        IsMoving = moving;
    }

    public String[] getAllData(){
        allData = new String[]{"ID: "+this.getId(),"Position: "+this.getPosition()+"","Load: "+this.getLoad()+"","Temperature: "+this.getTemperature()+"","Voltage: "+this.getVoltage()+""};
        return allData;
    }
}

File: src/movement/servo.py
from ax import Ax12
ax = Ax12()

class Servo(object):
    def __init__(self,id):
        self.id = id
        self.position = 0
        self.load = 0
        self.voltage = 0
        self.temperature = 0
        self.moving = 0

    def getId(self):
        return self.id

    def getPosition(self):
        return self.position

    def getLoad(self):
        return self.load

    def getVoltage(self):
        return self.voltage

    def getTemperature(self):
        return self.temperature

    def getMovingStatus(self):
        return self.moving

    def updateVariables(self):
        self.position = ax.readPosition(self.id)
        #print "position: ",position
        self.load = ax.readLoad(self.id)
        self.voltage = ax.readVoltage(self.id)
        self.temperature = ax.readTemperature(self.id)
        self.moving = ax.readMovingStatus(self.id)
